### ISSUE TITLE
Crash when trying to attach image to message

### ISSUE THREAD

**Issue by igor-sirotin** [status-im/status-react#20510]
# Problem

Reproduce steps:
1. Go to Status Community -> #mobile
2. Press "attach image" button

https://github.com/status-im/status-mobile/assets/25482501/43137227-21c5-4ee9-afd9-5940813d28a8

## Acceptance Criteria

No crash. Able to attach an image.

## Info

Device: iPhone 14 Pro
OS: iOS 17.5.1

App version: `2.29.20425 (20240617141826)`
App commit: c20a2da267973f7403655fe4524dfc9bd4a8fbcd
Node version: `StatusIM/v0.180.30/ios-arm64/go1.21.8`

**Comment by ilmotta** [comment]
For anyone going to investigate this issue, I couldn't reproduce on Android 13 on a real device and when pointing to the revision reported in this issue c20a2da267973f7403655fe4524dfc9bd4a8fbcd. Possibly an iOS-only bug.

@status-im/mobile-qa, whenever possible, could you check if you can reproduce this bug in other devices or is it only affecting this particular version of iOS?

**Comment by pavloburykh** [comment]
Hey @ilmotta! Not reproducing on my iPhone X, IOS 16.7.7

**Comment by VolodLytvynenko** [comment]
Hi. Can't reproduce it on iPhone 11 Pro max, IOS 17.5

https://github.com/status-im/status-mobile/assets/52490791/aa399fdc-444b-4b06-a834-899af023b9a6

**Comment by mariia-skrypnyk** [comment]
Hey hey @ilmotta !

Did the exact steps on nightly and no crash.
Tried on my iPhone 13 (17.5.1).

**Comment by igor-sirotin** [comment]
Thanks you for checking! 
I'm still able to reproduce it every time. Let me know if I should gather any details for you. 
I guess I can get a crash dump somehow?

**Comment by qoqobolo** [comment]
Hey @igor-sirotin, thanks for reporting! Do you have your mobile app synced with desktop?

**Comment by igor-sirotin** [comment]
I tried to limit photos access to 1 image and it's working. Also doesn't crash with no photo access.

<img width="200" src="https://github.com/status-im/status-mobile/assets/25482501/077009ef-8624-4637-8e6f-58a6411b0d3a"/>
<img width="200" src="https://github.com/status-im/status-mobile/assets/25482501/c6aeaaab-8068-44a8-8e4e-f3a085d083ad"/>
<img width="200" src="https://github.com/status-im/status-mobile/assets/25482501/ea4219fb-903d-4ac9-90f9-e8dc77393894"/>

But still crashes when going back to Full access to my 9k photos and 1,5k videos.

**Comment by igor-sirotin** [comment]
> Hey @igor-sirotin, thanks for reporting! Do you have your mobile app synced with desktop?

yes 🙂
But it doesn't seem related in any way

**Comment by qoqobolo** [comment]
> > Hey @igor-sirotin, thanks for reporting! Do you have your mobile app synced with desktop?
> 
> yes 🙂 But it doesn't seem related in any way

yeah I see now it's most likely related to permissions/some specific media in your gallery 👌

**Comment by churik** [comment]
for the record, also not reproducible for me
Device: iPhone 14 Pro
OS: iOS 17.5.1
![IMAGE 2024-06-24 17:02:55](https://github.com/status-im/status-mobile/assets/4557972/ec9b43fc-92d7-4e6e-a064-39baa28c0b7a)

I put `low-prio`so far, as no clue how to get the issue

**Comment by igor-sirotin** [comment]
lol, what is it in my photo library 😄

**Comment by ilmotta** [comment]
@igor-sirotin, let's give some more time for this issue to pop-up again from other users. I'm sorry to see we can't help yet.

The only other idea I can think of now is if you tried to develop the mobile app against your real device to better debug the error, but that's a whole lot to digest 😄 But if you want to venture yourself in the darkness of status-mobile let us know and we can try to help.

**Comment by igor-sirotin** [comment]
> let's give some more time for this issue to pop-up again from other users. I'm sorry to see we can't help yet.
> 
> The only other idea I can think of now is if you tried to develop the mobile app against your real device to better debug the error, but that's a whole lot to digest 😄 But if you want to venture yourself in the darkness of status-mobile let us know and we can try to help.

@ilmotta, no worries, thank you! 
I'll be ready to debug this at some point later 🙂

**Comment by flexsurfer** [comment]
we are loading images with 
```
[rn/image
      {:source {:uri (:uri item)}
       :style  (style/image window-width index)}]
       ```
       so most likely something wrong with the image, corrupt or something? not supported format might be

**Comment by flexsurfer** [comment]
hey @igor-sirotin please try this one https://github.com/status-im/status-mobile/pull/20574

**Comment by flexsurfer** [comment]
thanks Igor, so now we know it's not the image rendering, the error happens in the second request for `cameraroll/get-photos`

**Comment by J-Son89** [comment]
@flexsurfer is this issue still present?

**Comment by ilmotta** [comment]
I'm removing the issue from the 2.30 milestone due to the difficulty in reproducing, in combination with our time constraints for the upcoming 2.30 release.

Thank you QA team for trying to promptly investigate the bug.

@igor-sirotin, we will still keep a close eye on this issue if more users report this bug because crashes are no fun. Thanks for reporting and trying to help us figure out the root cause.

**Comment by ilmotta** [comment]
Update: another user reported in Sep 27 a similar bug. Reference https://www.notion.so/Testing-it-1118f96fb65c808c97bcc04c1b3ce214

**Comment by VolodLytvynenko** [comment]
Reproduced this issue on: iPhone 16 IOS: 18.0
https://app.eu-central-1.saucelabs.com/tests/0a5612ad034445c7b011636ce359a831

logs:
[SauceLabs_Log_iPhone_16_18_real_2024-10-04_03-58.txt](https://github.com/user-attachments/files/17259625/SauceLabs_Log_iPhone_16_18_real_2024-10-04_03-58.txt)

**Comment by VolodLytvynenko** [comment]
@ilmotta @churik I've moved this issue to the 2.32 Milestone because I think it's device or OS specific. I've tested it on 5 iOS devices and was only able to reproduce it on an iPhone 16 on iOS 18, although the user reported it on a different device. I'm mentioning it here in case it needs to be included in 2.31. Feel free to move it to 2.31 if you think it has a higher priority to be fixed

**Comment by ilmotta** [comment]
@VolodLytvynenko I guess most people, like myself, outside the QA team can't access SauceLabs, so we can't watch the video.

If we solve this issue, it will definitely be for 2.32 or beyond. Thanks for updating the issue.

**Comment by flexsurfer** [comment]
@VolodLytvynenko  maybe the problem is not in the device or OS, but in the data or some kind of image, or there is probably an edge case, could you experiment with the device there you have crash and try to avoid it, idk, change gallery images etc

**Comment by ulisesmac** [comment]
@VolodLytvynenko Probably you can privately share with us the photo you trid to send as a file and we can try replicating the issue, or you could try the same photo on different devices.

**Comment by VolodLytvynenko** [comment]
> [@VolodLytvynenko](https://github.com/VolodLytvynenko) Probably you can privately share with us the photo you trid to send as a file and we can try replicating the issue, or you could try the same photo on different devices.

@ulisesmac Unfortunately, I can't reproduce this issue anymore and can't share the photo because I was using a Sauce Labs device when the crash occurred. Currently, I am unable to reproduce the issue on the same device and iOS version. I can only share the Sauce Labs recording, but there doesn't seem to be anything unusual in my steps:

1. Finished onboarding (1:03)
2. Took a photo using the device camera (1:15)
3. Sent the photo in a 1-1 chat, which caused the crash (2:21)
 
https://github.com/user-attachments/assets/80a3af37-163a-4222-9001-96aa05f3e717

**Comment by VolodLytvynenko** [comment]
Here are some additional logs that Sauce Labs can provide
https://www.dropbox.com/scl/fi/fdzhkxn1ijfjfn1lb1cm9/device-1.log?rlkey=833taj3sp8xuqhimi1bhbmo0c&st=ra76vvxe&dl=0

**Comment by flexsurfer** [comment]
hm @VolodLytvynenko it seems like it is another issue, in the current task crash happened when trying to attach image, not send

**Comment by flexsurfer** [comment]
they might be related but we don't know that, I would register it as a separate issue

**Comment by igor-sirotin** [comment]
I still can reproduce every time. 
If someone wants to help me set up debugging with my phone, feel free to reach out to me, I'can be your hands to do debugging 🙂

**Comment by churik** [comment]
I don't think that we have a capacity for it in 2.32, given the desired release cut date 
Deferring the issue, please be free to move it bak if you think there is a room for it @ilmotta

**Comment by ilmotta** [comment]
Heads up @churik @igor-sirotin: this issue is getting old already and since it was first reported, none of us could replicate the problem consistently. @mariia-skrypnyk and the rest of the core team discussed today about what to do with this issue. As already mentioned in previous comments, a mobile dev would need to pair with @igor-sirotin to hopefully spot the pesky bug by connecting to his real device.

Due to other priorities for 2.33, we aren't going to be able to focus on this issue in the next couple weeks and unclear if we will be able to work on it for 2.33.

But having said that, maybe somebody with macOS can jump in from @status-im/mobile-core or @status-im/wallet-mobile-devs to pair with @igor-sirotin. That'd be fantastic!

**Comment by igor-sirotin** [comment]
@ilmotta thank you, I'm ready to help when needed.
Just in case, tested it now with 2.32, the issue is still present

**Comment by churik** [comment]
it seems we're not taking it to 2.33 as well, I'll try to prioritize it for 2.34

**Comment by churik** [comment]
@ilmotta leave it to you to prioritize

**Comment by ilmotta** [comment]
Aligned with @ulisesmac that during the 2.34 development cycle he can give a shot to this bug while @igor-sirotin shares his screen.

Igor would likely need to set-up his dev environment to be able to connect to his real device and have the emulator running so we can debug (we can help you with that of course).

I cannot develop in iOS, but another developer could join Ulises, maybe @Parveshdhull (?) and they can timebox the investigation to 1-3h and if nothing useful comes out of the investigation we will have to close this issue as _not planned_.

Up to you folks to figure out the best date/time. Most likely not now because we are deep in the release 2.33.

Please, thumbs up if you think this sounds like a reasonable plan, otherwise I think we should close this issue as _not planned_ because it's rotting 🍅

Thank you in advance 🙏🏼

**Comment by ulisesmac** [comment]
I can develop on iOS too, so I think I can debug it with @igor-sirotin. If we face problems, we could ask for @Parveshdhull 's help :)
